sdk/monitor/azure-monitor-opentelemetry-exporter/tests/test_base_exporter.py

```diff
@@ -81,49 +87,61 @@ def test_constructor(self):
         self.assertEqual(base._timeout, 10)
         self.assertEqual(base._api_version, "2021-02-10_Preview")
         self.assertEqual(base._storage_min_retry_interval, 100)
+        self.assertEqual(base._storage_path, "test/path")
+
 
+    def test_transmit_from_storage_failed(self):
+        exporter = BaseExporter()
+        exporter.storage = mock.Mock()
+        blob_mock = mock.Mock()
+        blob_mock.lease.return_value = True
+        envelope_mock = {"name":"test","time":"time"}
+        blob_mock.get.return_value = [envelope_mock]
+        exporter.storage.gets.return_value = [blob_mock]
+        with mock.patch.object(AzureMonitorClient, 'track') as post:
+            post.return_value = TrackResponse(
+                items_received=1,
+                items_accepted=1,
+                errors=[],
+            )
+            exporter._transmit_from_storage()
+        exporter.storage.gets.assert_called_once()
+        blob_mock.lease.assert_called_once()
+        blob_mock.delete.assert_called_once()
 
-    @unittest.skip("transient storage")
     def test_transmit_from_storage_failed_retryable(self):
-        envelopes_to_store = [x.as_dict() for x in self._envelopes_to_export]
-        self._base.storage.put(envelopes_to_store)
-        # Timeout in HTTP request is a retryable case
-        with mock.patch("requests.Session.request", throw(requests.Timeout)):
-            self._base._transmit_from_storage()
-        # File would be locked for 1 second
-        self.assertIsNone(self._base.storage.get())
-        # File still present
-        self.assertGreaterEqual(len(os.listdir(self._base.storage._path)), 1)
-
-    @unittest.skip("transient storage")
-    def test_transmit_from_storage_failed_not_retryable(self):
-        envelopes_to_store = [x.as_dict() for x in self._envelopes_to_export]
-        self._base.storage.put(envelopes_to_store)
-        with mock.patch("requests.Session.request") as post:
-            # Do not retry with internal server error responses
-            post.return_value = MockResponse(400, "{}")
-            self._base._transmit_from_storage()
-        self._base.storage.get()
-        # File no longer present
-        self.assertEqual(len(os.listdir(self._base.storage._path)), 0)
+        exporter = BaseExporter()
+        exporter.storage = mock.Mock()
+        blob_mock = mock.Mock()
+        blob_mock.lease.return_value = True
+        envelope_mock = {"name":"test","time":"time"}
+        blob_mock.get.return_value = [envelope_mock]
+        exporter.storage.gets.return_value = [blob_mock]
+        with mock.patch("azure.monitor.opentelemetry.exporter.export._base._is_retryable_code"):
+            with mock.patch.object(AzureMonitorClient, 'track', throw(HttpResponseError)):
```

Comment:
Can't we combine these 2 with statements?

Reply:
I'm not sure of how the syntax of that looks and don't think it's a big deal.

Reply:
Not a big deal, but u can just use a comma. https://stackoverflow.com/questions/893333/multiple-variables-in-a-with-statement